omnigibson/utils/usd_utils.py

```diff
@@ -808,40 +1267,33 @@ def get_mesh_volume_and_com(mesh_prim, world_frame=False):
     return volume, com
 
 
-def check_extent_radius_ratio(mesh_prim):
+def check_extent_radius_ratio(geom_prim, com):
     """
-    Checks if the min extent in world frame and the extent radius ratio in local frame of @mesh_prim is within the
+    Checks if the min extent in world frame and the extent radius ratio in local frame of @geom_prim is within the
     acceptable range for PhysX GPU acceleration (not too thin, and not too oblong)
 
     Ref: https://github.com/NVIDIA-Omniverse/PhysX/blob/561a0df858d7e48879cdf7eeb54cfe208f660f18/physx/source/geomutils/src/convex/GuConvexMeshData.h#L183-L190
 
     Args:
-        mesh_prim (Usd.Prim): Mesh prim to check
+        geom_prim (GeomPrim): Geom prim to check
+        com (np.array): Center of mass of the mesh. Obtained from get_mesh_volume_and_com
 
     Returns:
         bool: True if the min extent (world) and the extent radius ratio (local frame) is acceptable, False otherwise
     """
-    mesh_type = mesh_prim.GetPrimTypeInfo().GetTypeName()
+    mesh_type = geom_prim.prim.GetPrimTypeInfo().GetTypeName()
     # Non-mesh prims are always considered to be within the acceptable range
     if mesh_type != "Mesh":
         return True
 
-    trimesh_mesh_world = mesh_prim_to_trimesh_mesh(
-        mesh_prim, include_normals=False, include_texcoord=False, world_frame=True
-    )
-    min_extent = trimesh_mesh_world.extents.min()
+    extent = geom_prim.extent
+    min_extent = extent.min()
     # If the mesh is too flat in the world frame, omniverse cannot create convex mesh for it
     if min_extent < 1e-5:
         return False
 
-    trimesh_mesh = mesh_prim_to_trimesh_mesh(
-        mesh_prim, include_normals=False, include_texcoord=False, world_frame=False
-    )
-    if not trimesh_mesh.is_volume:
-        trimesh_mesh = trimesh_mesh.convex_hull
-
-    max_radius = trimesh_mesh.extents.max() / 2.0
-    min_radius = trimesh.proximity.closest_point(trimesh_mesh, np.array([trimesh_mesh.center_mass]))[1][0]
+    max_radius = extent.max() / 2.0
+    min_radius = np.min(np.linalg.norm(geom_prim.points - com, axis=-1), axis=0)
```

Comment:
hmm I think `min_radius` is computed differently from before now. Previously, min_radius is the closest point on any triangle to the CoM (doesn't have to be one of the vertices)

Reply:
@cgokmen

Reply:
hm this is true. it's actually almost guaranteed to not be one of the vertices. lets leave this in for now, @hang-yin please create a ticket for a follow up, P0 this time, starter issue. This is not a showstopper we should fix after merge to avoid more changes to the huge PR